src/Shared/WebEncoders/WebEncoders.cs

```diff
@@ -69,9 +74,11 @@ public static byte[] Base64UrlDecode(string input, int offset, int count)
 
 #if NET9_0_OR_GREATER
         // Legacy behavior of Base64UrlDecode supports either Base64 or Base64Url input.
-        // If it doesn't have + or /, it can be treated as Base64Url.
+        // If it has a - or _, or if it doesn't have + or /, it can be treated as Base64Url.
+        // Searching for any of them allows us to stop the search as early as we know Base64Url should be used.
```

Comment:
```suggestion
        // Searching for any of them allows us to stop the search as early as we know whether Base64Url should be used.
```

Reply:
It sounds like finding one of the characters lets us make a decision, rather than telling us that Base64Url should be used, though I may well have misunderstood.

Reply:
I think we're saying the same thing. By providing all of the characters, we can stop the moment we know Base64Url can be used. If we only provided + and /, we'd have to search the whole thing, as we'd only know Base64Url could be used once the entire input is searched.

Reply:
In that case, I think "whether" helps clarify that, though I don't feel strongly about it.  In particular, I'll understand if you'd rather not battle CI again. 😆